src/libecalc/presentation/yaml/yaml_types/emitters/yaml_venting_emitter.py

```diff
@@ -55,19 +88,23 @@ def component_type(self):
         validate_default=True,
     )
 
-    emission: YamlVentingEmission = Field(
+    type: YamlVentingType = Field(
```

Comment:
Separating the two types into different classes with a discriminator will probably give better errors/help while editing. It could also clean up the validators and remove some checks. I.e. get_emission_rates could be implemented for the two classes where each contained either `_get_oil_type_emissions` or `_get_direct_type_emissions`

Reply:
@jsolaas : The two types are separated to different classes. Please have a look.